legacy_packages/sdk/src/evm/core/classes/transactions.ts

```diff
@@ -449,13 +452,18 @@ export class Transaction<
     }
 
     const gasOverrides = await this.getGasOverrides();
-    const overrides: CallOverrides = { ...gasOverrides, ...this.overrides };
+    const overrides: CallOverridesWithExtraGas = { ...gasOverrides, ...this.overrides };
 
     // First, if no gasLimit is passed, call estimate gas ourselves
     if (!overrides.gasLimit) {
       overrides.gasLimit = await this.estimateGasLimit();
     }
 
+    if (overrides.extraGas) {
+      const originalLimit = await overrides.gasLimit;
```

Comment:
do we need to await again here?

Reply:
We technically shouldn't but TS requires it because `gasLimit` type on the overrides object is `Promise<BigNumber> | BigNumber`